### DISCUSSION TITLE
Max number of access keys?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [minio/minio#17399] [Q&A] [answered]
I would like to utilize single-bucket design, and have access keys scoped by the object path (e.g. `user_a` can only access files at `singlebucket/users/user_a/**`.

Is there any limit to the number of max keys that I can create? I would put the permissions within the access key, and give the user direct access to the minio s3 apiq for consumption.

**Comment by harshavardhana** [comment, marked as answer]
The limit is as big as your disk can handle.

**Reply by danthegoodman1** [reply]
Will there be any performance penalty at say 100k accounts?

**Reply by harshavardhana** [reply]
Depends on your setup.

**Reply by danthegoodman1** [reply]
So assuming there are enough resources, it should remain fast? Thanks.

**Reply by harshavardhana** [reply]
Can you define resources? How many drives in the setup ?

You want architecture guidance please come to us via subscription https://min.io/pricing to get a more professional advice.

**Reply by danthegoodman1** [reply]
Nothing concrete, just wanted to make sure there weren’t any hard numerical limits like AWS limiting to 5k IAM accounts :)
